### PR TITLE
Add warning about sidecar egress

### PR DESCRIPTION
**Pull request by frankbu**
Fixes https://github.com/istio/istio/issues/12237

### CODE REVIEW

**Comment by linsun** [comment]
/lgtm
a little unfortunate user has to do this manually but ok with it being a temp issue till mixer v2 is in place.

cc @geeknoid @louiscryan

**Comment by louiscryan** [comment]
/lgtm

agreed about the pain here. Is there a tracking issue for resolving

File: networking/v1alpha3/sidecar.proto [hunk 1]
```diff
@@ -289,14 +289,22 @@ message IstioEgressListener {
...
   //
   // NOTE: Only services and configuration artifacts exported to the sidecar's
   // namespace (e.g., `exportTo` value of `*`) can be referenced.
   // Private configurations (e.g., `exportTo` set to `.`) will
   // not be available. Refer to the `exportTo` setting in `VirtualService`,
   // `DestinationRule`, and `ServiceEntry` configurations for details.
+  //
+  // **WARNING:** The list of egress hosts in a `Sidecar` must also include
+  // the Mixer control plane services if they are enabled. Envoy will not
+  // function properly otherwise. For example add host
```
Comment (rshriram) [inline networking/v1alpha3/sidecar.proto]: I would say that if telemetry is enabled, add istio-system/telemetry.. same for policy. Else it sounds like you always need this.

Comment (howardjohn) [reply networking/v1alpha3/sidecar.proto]: Agreed, "function properly" is a bit vague. Your app won't break, since you can still connect with Pilot

**Comment by xeviknal** [comment]
Is this warning still valid on istio 1.5?

**Comment by howardjohn** [comment]
Only if you use mixer

**Pull request merged**